### PR TITLE
Grant Bucket Access [WIP]

### PR DESCRIPTION
**Pull request by YiannisGkoufas**
Take on #51

### CODE REVIEW

File: cmd/minio-cosi-driver/internal/minio/client.go [hunk 1]
```diff
@@ -52,23 +58,26 @@ func NewClient(ctx context.Context, minioHost, accessKey, secretKey string) (*C,
...
 
-		cl, err := min.New(host.Host, &min.Options{
+		client, err := min.New(host.Host, &min.Options{
 			Creds:  credentials.NewStaticV4(accessKey, secretKey, ""),
 			Secure: secure,
 		})
 		if err != nil {
 			errChan <- err
 		}
-		_, err = cl.BucketExists(ctx, uuid.New().String())
```
Comment (wlan0) [inline cmd/minio-cosi-driver/internal/minio/client.go:67]:
```go
		adminClient, err := madmin.New(host.Host, accessKey, secretKey, secure)
		if err != nil {
			errChan <- errors.Wrap(err, "Connection to MinIO as admin failed")
			return
		}
```

File: cmd/minio-cosi-driver/internal/minio/client.go [hunk 2]
```diff
@@ -52,23 +58,26 @@ func NewClient(ctx context.Context, minioHost, accessKey, secretKey string) (*C,
...
 	go func() {
 		klog.V(3).InfoS("Connecting to MinIO", "endpoint", host.Host)
 
-		cl, err := min.New(host.Host, &min.Options{
+		client, err := min.New(host.Host, &min.Options{
 			Creds:  credentials.NewStaticV4(accessKey, secretKey, ""),
 			Secure: secure,
 		})
 		if err != nil {
 			errChan <- err
```
Comment (wlan0) [inline cmd/minio-cosi-driver/internal/minio/client.go:71]:
```go
errChan <- err
return
```

File: cmd/minio-cosi-driver/internal/minio/client.go [hunk 3]
```diff
@@ -52,23 +58,26 @@ func NewClient(ctx context.Context, minioHost, accessKey, secretKey string) (*C,
...
 		}
-		_, err = cl.BucketExists(ctx, uuid.New().String())
+		_, err = client.BucketExists(ctx, uuid.New().String())
```
Comment (wlan0) [inline cmd/minio-cosi-driver/internal/minio/client.go:73]: use admin client here

Comment (wlan0) [reply cmd/minio-cosi-driver/internal/minio/client.go:73]: for the admin client, call ServerInfo and check that err is nil

**Comment by YiannisGkoufas** [comment]
Closing it to move WIP in https://github.com/kubernetes-sigs/cosi-driver-minio.git

**Pull request closed without merging**